resources/abi/README.md

```diff
@@ -0,0 +1,74 @@
+# ABI change monitoring
+
+When the OpenAssetIO C++ ABI changes, we need to know about it to inform
+our release notes and semver versioning.
+
+This is aided (though not solved!) by automated testing against a
```

Comment:
[Question] How not solved? Like, there's an outside but not realistic shot at libabigail missing things, or it misses a lot of common things?

The exclamation mark scares me here, it's saying to me that I can't trust this tool and I need to do additional work myself, and I think, what do I as an implementor need to be aware of? There is a point where if we still have to be vigilant, the overhead of the tooling isn't worth it at all.

Reply:
Oh yes we still have to be vigilant.  libabigail, from what I've seen, won't miss any changes in the binary, but they won't necessarily be breaking changes, so we need manual effort to decode the output and decide if and how the change is breaking.  

Most troubling is that changes in template classes or inline functions (i.e. things defined just in headers) won't be picked up.  This is eased as long as it's used in the binary.  E.g. I tried messing with `BatchElementError`, which is only defined in a header, but since it's used in the binary, the ABI change was picked up by libabigail.  

Note that using the command-line switch to point `abidiff` at the headers (`--headers-dir-1`) doesn't cause it to check their ABI, it's only used to filter the output from scanning the binary.

Also, although the docs you highlight is focused on ABI, I did have in mind things that affect versioning more generally, which are not necessarily ABI breaks - e.g. trait changes, changes to the documented contract, new non-breaking features.

Reply:
Do you think these specific points should be noted in this document? How else is someone meant to know what to look out for?

Reply:
I like the idea of a Limitations section :+1:

Reply:
Added a section in bad49c8d.

Reviewed over-the-shoulder by @elliotcmorris.